src/osx/window_osx.cpp

```diff
@@ -2629,16 +2629,23 @@ wxSize wxWindowMac::OSXMakeDPIFromScaleFactor(double scaleFactor)
 }
 
 wxSize wxWindowMac::GetDPI() const
+{
+    return OSXMakeDPIFromScaleFactor(GetDPIScaleFactor());
+}
+
+// on mac ContentScale and DPIScale are identical
+double wxWindowMac::GetDPIScaleFactor() const
 {
     double scaleFactor;
     if ( wxNonOwnedWindow* tlw = MacGetTopLevelWindow() )
         scaleFactor = tlw->GetContentScaleFactor();
     else
         scaleFactor = wxOSXGetMainScreenContentScaleFactor();
```

Comment:
Should we ask the window display for its scale factor here?

Of course, normally we should always have the containing TLW anyhow, so maybe it's not worth it...

Reply:
we should ask the TLW as there we have the 'true' value, the way it is handled by the OS. If there isn't a tlw, there isn't a absolute position either, so I think using the main screen's scale is as good or bad as any other. Or did I not understand you correctly ?

Reply:
I'm not sure, could we have a position but no TLW because it wasn't shown (at this position) yet?

Reply:
Actually, why don't we just do the same thing (or just call) `GetContentScaleFactor()` here? I don't understand why is there the difference between the two, as the comment says, they should return the same thing?

Reply:
yes, they should return the same thing, but I haven't built in a fallback in case there's no tlw, so let's have both go through the same route and I'll add that condition to the code